scripts/base/packet-protocols/ethernet/main.zeek

```diff
@@ -0,0 +1,26 @@
+module PacketAnalyzer::Ethernet;
+
+export {
+	## IEEE 802.2 SNAP analyzer
+	const snap_analyzer: PacketAnalyzer::Tag &redef;
```

Comment:
I had the same question earlier. These are basically here as a placeholder for future work.

Reply:
Ethernet frames come in four different frame formats (see https://en.wikipedia.org/wiki/Ethernet_frame#Types). Ethernet II seems to be the most common and is what zeek handled so far. I think the other frame formats are best supported by adding new analyzers for them (especially as LLC and SNAP are used in other places as well). The differentiation between the frame formats is not solely based on an identifier one can use for dispatching. Thus the Ethernet analyzer takes care of detecting the frame type and allows to configure analyzers for the exotic ones here.

Reply:
Ack, my main point was that we're back to hardcoding some next-stage analyzers in C++ here. Maybe that's the only way to do it, but it would be cool if we could get them to take-over through a normal identifier mapping. But I see that "data" doesn't move for them, which might be what's preventing this? 

If we keep this, do we need the `const` definitions in script-land? they are `&redef` but would anybody really want to change them?

Reply:
Hmm... I am a bit confused. The analyzers are not hardcoded. It's the identification that is hardcoded. You're right that `data` isn't moved but that's not an issue. The problem is that the ether types up to 1500 are ambiguous. One needs to look at the first bytes of the payload to distinguish further. As there are no analyzers for LLC and SNAP so far, the `const`s will be redefed once someone comes up with an analyzer.

Reply:
I'm confused now. If we don't have the analyzers yet (none of them?), do we want this code at all? 

And what I meant above is: can we use a ConfigEntry to do the dispatching, like:

```
PacketAnalyzer::ConfigEntry($parent=PacketAnalyzer::ANALYZER_ETHERNET, $identifier=SOME_IDENT_FOR_SNAP, $analyzer=PacketAnalyzer::ANALYZER_ETHERNET_SNAP)
```

Reply:
Actually, that's the approach I implemented [at first](https://github.com/zeek/zeek/pull/1046/commits/01ac7b0f3f9375f783ed97f94b6788219a61e2a7#diff-a0a6f2b9079af21f727d4ac8cde8f6e2). However, that approach required me to define "magic ether types" (1501-1503) to be used for this purpose. Instead of using these magic values, now one can configure the analyzers like this:
```
redef PacketAnalyzer::Ethernet::snap_analyzer = PacketAnalyzer::ANALYZER_ETHERNET_SNAP;
```
So it's a trade-off between an additional way of configuration and using magic values.

Reply:
This thread was never fully-resolved. Should we keep with the magic numbers, or add placeholders for analyzers that may not exist?